src/components/__tests__/TestUtils.ts

```diff
@@ -0,0 +1,119 @@
+import { expect } from "chai";
+import * as Enzyme from "enzyme";
+
+import { LibraryData } from "../../interfaces";
+
+export const testLibrary1: LibraryData = {
+  uuid: "UUID1",
+  basic_info: {
+    "name": "Test Library 1",
+    "short_name": "lib1",
+    "description": undefined
+  },
+  urls_and_contact: {
+    "authentication_url": "auth1",
+    "contact_email": "email1",
+    "opds_url": "opds1",
+    "web_url": "web1"
+  },
+  stages: {
+    "library_stage": "production",
+    "registry_stage": "testing"
+  },
+  areas: {
+    "focus": ["10025 (NY)", "Stratford (ON)", "33154 (FL)"],
+    "service": []
+  }
+};
+
+export const testLibrary2: LibraryData = {
+  uuid: "UUID2",
+  basic_info: {
+    "name": "Test Library 2",
+    "short_name": "lib2",
+  },
+  urls_and_contact: {
+    "authentication_url": "auth2",
+    "contact_email": "email2",
+    "opds_url": "opds2",
+    "web_url": "web2"
+  },
+  stages: {
+    "library_stage": "testing",
+    "registry_stage": "cancelled"
+  },
+  areas: {
+    "focus": [],
+    "service": []
+  }
+};
+
+/**
+* modifyLibrary()
+* @param {LibraryData} baseLibrary - the library to be modified
+* @param {{[key: string]: string | string[]}} newData - key-value pairs with which to update the library
+* @param {string} [category] - the category that the new values should go into.  This isn't necessary if we're
+modifying the value of a key that already exists in the library; modifyLibrary will figure it out.
+It's only necessary if we're adding a new key.
+*/
+export const modifyLibrary = (baseLibrary: LibraryData, newData: {[key: string]: string | string[]}, category?: string): LibraryData => {
+  let categories = Object.keys(baseLibrary);
+  let updatedLibrary = {...baseLibrary};
+
+  Object.entries(newData).forEach((pair) => {
+    let categoryToUpdate = category || null;
+    let newValue = {};
+    let [key, value] = pair;
+    if (!categoryToUpdate) {
+      let allCategories = Object.keys(baseLibrary);
+      categoryToUpdate = allCategories.find((cat) => Object.keys(baseLibrary[cat]).includes(key));
+    }
+    newValue[key] = value;
+    let updatedCategory = { ...(updatedLibrary[categoryToUpdate] || baseLibrary[categoryToUpdate]), ...newValue };
```

Comment:
This is confusing. Isn't `updatedLibrary` a copy of `baseLibrary` from above? Neither `updatedLibrary` or `baseLibrary` get modified until you call this here.

Reply:
I guess for the first iteration it won't matter but the rest of the loop is fine.

Reply:
Actually, since the `categoryToUpdate` value will always be one of the keys from the `baseLibrary` and `updatedLibrary` is created from it, then can't you only just use `updatedLibrary` here?

Reply:
You can, but I think that makes it a little harder to read.  See how it looks to you...

Reply:
Looks more readable because it makes sense where the value is coming from.